services/motion/builtin/move_request.go

```diff
@@ -202,68 +325,52 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 
 	for visSrvc, cameraNames := range mr.obstacleDetectors {
 		for _, camName := range cameraNames {
-			mr.logger.Debugf(
-				"proceeding to get detections from vision service: %s with camera: %s",
-				visSrvc.Name().ShortName(),
-				camName.ShortName(),
-			)
-
-			// get detections from vision service
-			detections, err := visSrvc.GetObjectPointClouds(ctx, camName.Name, nil)
+			// get the current position of the base which we will use to transform the detection into world coordinates
+			currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
 			if err != nil {
 				return state.ExecuteResponse{}, err
 			}
+			// make sure the currentPosition is in the world frame
+			if currentPosition.Parent() != referenceframe.World {
+				currentPosition, err = mr.fsService.TransformPose(ctx, currentPosition, referenceframe.World, nil)
```

Comment:
I think this should remain.

> The FS service does not know where the SLAM map is

This is true, but I feel like this doesn't matter here. If are using a SLAM based localizer, our returned PIF will have parent as world

> it does not know what GPS coordinates we are computing pose relative to, so this call will probably return a useless (wrong) value.

I think we do know what we are computing pose relative to since the movement sensor based localizer knows its origin. Here our PIF will have parent as movement sensor so we need to transform to be in the world frame so that we know where the base frame is. Here however, I assume that the base and world frame are co-incident.

Reply:
The trouble is this is using the frame system service, which will be querying a different frame system than what is actually used for solving. It's unknown what parentage/transforms will have been put between the base and world, hence this call may return entirely wrong information.

In the case where the parent of the PiF is not World, the better thing to do would be to get the CurrentInputs from the fs service, but use the framesystem in the `motionplan.PlanRequest` associated with the `moveRequest` to transform to `world`, as that is the framesystem that was used to generate the plan.